### PR TITLE
Examples build fixes

### PR DESCRIPTION
**Pull request by kpal81xd**
Refers to bugs in #6030

- Moved prebuild step to inside rollup
- Fixed bug with production build using bundled modules

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: examples/rollup.config.js [hunk 1]
```diff
@@ -51,59 +59,37 @@ const staticFiles = [
...
-    } else {
-        // This can be both UMD/ESM as a single file
-        const entryPoint = ENGINE_PATH.split("/").pop();
-        const dest = 'dist/iframe/ENGINE_PATH/' + entryPoint;
-        staticFiles.push({ src, dest });
+        return [{ src: srcDir, dest }];
     }
+
+    // packed module builds
+    ENGINE_PATH.split("/").pop();
```
Comment (kungfooman) [inline examples/rollup.config.js:85]: This looks like a statement that just does nothing so far? If you still work on it, please mark it as draft until ready for review :detective:

File: examples/rollup.config.js [hunk 2]
```diff
@@ -138,8 +124,8 @@ function watch(plugin, src) {
 /**
  * This plugin copies static files from source to destination.
  *
- * @param {staticFiles} targets - Array of source and destination objects.
- * @returns {RollupPlugin} The plugin.
+ * @param {STATIC_FILES} targets - Array of source and destination objects.
+ * @returns {RollupPlugin} - The plugin.
```
Comment (kungfooman) [inline examples/rollup.config.js:128]: `@returns` without dash

File: examples/rollup.config.js [hunk 3]
```diff
@@ -12,19 +12,27 @@ import replace from '@rollup/plugin-replace';
...
+// util functions
+import { isModuleWithExternalDependencies } from './utils.mjs';
+
 /** @typedef {import('rollup').RollupOptions} RollupOptions */
 /** @typedef {import('rollup').Plugin} RollupPlugin */
 
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 
+const NODE_ENV = process.env.NODE_ENV ?? '';
+const ENGINE_PATH = !process.env.ENGINE_PATH && NODE_ENV === 'development' ? '../src/index.js' : process.env.ENGINE_PATH ?? '';
```
Comment (kungfooman) [inline examples/rollup.config.js:28]: Line length limit of 100 - would also support reading :sweat_smile:

File: examples/rollup.config.js (hunk 1 above)
Comment (kpal81xd) [reply examples/rollup.config.js:85]: That line should have been removed 😅

Comment (kpal81xd) [inline examples/rollup.config.js:85]:
```suggestion
```

File: examples/utils.mjs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * @param {string} content - The file content to test.
+ * @returns {boolean} Whether content is a module.
+ */
+export function isModuleWithExternalDependencies(content) {
+    const a = regexpExportStarFrom.test(content);
+    const b = regexpExportFrom.test(content);
+    const c = regexpImport.test(content);
+    // console.log('isModuleWithExternalDependencies', { a, b, c });
+    return a || b || c;
+}
```
Comment (willeastcott) [inline examples/utils.mjs:22]:
```suggestion
const regexPatterns = [
    /^\s*export\s*\*\s*from\s*.+\s*;\s*$/gm,
    /^\s*export\s*{.*}\s*from\s*.+\s*;\s*$/gm,
    /^\s*import\s*.+\s*;\s*$/gm,
];

/**
 * Checks if the provided content matches any of a set of patterns indicative of an ES Module with external dependencies.
 * Patterns checked include certain export and import statement formats.
 * 
 * @param {string} content The file content to test.
 * @returns {boolean} Whether the content is likely an ES Module with external dependencies.
 * @example
 * isModuleWithExternalDependencies(`
 *     // Testing variants:
 *     export * from './index.mjs';
 *     export { Ray } from './core/shape/ray.js';
 *     import './polyfill/OESVertexArrayObject.js';
 * `);
 */
export const isModuleWithExternalDependencies = (content) => regexPatterns.some(pattern => pattern.test(content));
```

**Review by willeastcott — APPROVED** [review]

File: examples/utils.mjs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const regexPatterns = [
+    /^\s*export\s*\*\s*from\s*.+\s*;\s*$/gm,
```
Comment (marklundin) [inline examples/utils.mjs:2]: Do we need to worry about dynamic imports? Also should we allow statements without semi-colons? I guess we enforce this on the linting level?

**Pull request merged**